File: examples/test_file.js
// Example JavaScript file with issues for CodeCure testing

function problemFunction() {
    console.log("Debug message");  // Should suggest proper logging
    
    let value = "5";
    if (value == 5) {  // Should suggest using === instead
        console.log("Values match");
    }
    
    // TODO: Add input validation
    
    return value;
}

function anotherFunction() {
    console.log("Another debug statement");
    
    let result = problemFunction();
    
    // This line has trailing spaces    
    
    if (result != undefined) {  // Should suggest using !== instead
        return result;
    }
}

// Very long line that exceeds reasonable length and should trigger a warning about line length in the code
let veryLongVariableName = "This is a very long line that should trigger a line length warning from CodeCure";

anotherFunction();

File: examples/test_file.py
# Example Python file with various issues for testing CodeCure

import os

def problematic_function():
    # This function has several issues that CodeCure should detect
    
    print("Debug message")  # Should suggest using logging
    
    try:
        result = 10 / 0
    except:  # Bare except clause
        print("An error occurred")
    
    # TODO: Implement proper error handling
    
    unused_variable = "this is not used"
    
    very_long_line_that_exceeds_reasonable_length_and_should_trigger_a_warning_about_line_length_issues = "too long"
    
    return result  # This will cause an error since result might not be defined


def another_function():
    print("Another debug statement")
    
    # Some trailing whitespace here:   
    
    if True == False:  # Could suggest using 'is' for boolean comparison
        pass


if __name__ == "__main__":
    problematic_function()
    another_function()